core/common/src/test/java/alluxio/master/PollingMasterInquireClientTest.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master;
+
+import static org.junit.Assert.fail;
+
+import alluxio.Constants;
+import alluxio.conf.ConfigurationBuilder;
+import alluxio.exception.status.UnavailableException;
+import alluxio.network.RejectingServer;
+import alluxio.retry.CountingRetry;
+import alluxio.util.network.NetworkAddressUtils;
+
+import org.junit.Test;
+
+import java.net.InetSocketAddress;
+import java.net.ServerSocket;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Unit tests for {@link PollingMasterInquireClient}.
+ */
+public class PollingMasterInquireClientTest {
+  @Test(timeout = 10000)
+  public void pollRejectingDoesntHang() throws Exception {
+    int port = PortRegistry.INSTANCE.reservePort();
+    RejectingServer s = new RejectingServer(port);
+    s.start();
+    List<InetSocketAddress> addrs = Arrays.asList(
+        new InetSocketAddress(NetworkAddressUtils.getLocalHostName(Constants.SECOND_MS), port));
+    PollingMasterInquireClient client = new PollingMasterInquireClient(addrs,
+        () -> new CountingRetry(0), new ConfigurationBuilder().build());
+    try {
+      client.getPrimaryRpcAddress();
+      fail("Expected polling to fail");
+    } catch (UnavailableException e) {
+      // Expected
+    }
+  }
+
+  /**
+   * Example of how polling a regular server socket will hang.
+   */
+//  @Test
```

Comment:
Annotate with ignore or remove comment

Reply:
I just don't like leaving the commented line there. The comment above doesn't really explain why. I don't think having an `@Ignore` necessarily implies that it can be run as a real unit test. If you specify the reason `@Ignore("If run the test will hang. Used to verify that creating a serversocket and closing")` it can easily indicate why we don't run the test.

Either it would make me happy if we had a good explanation about the behavior and why we don't run the test so that the next person that comes along understands why it was put here in the first place

Reply:
I agree it's confusing, and I don't see it providing much value in the future. It's already served the purpose of indicating why this PR was needed. I'll remove it